### DISCUSSION TITLE
Is it possible to pass timestamps also with nanosecond resolution to loki?

### DISCUSSION THREAD

**Discussion by thgau** [tkowalcz/tjahzi#95] [Q&A] [answered]
I use the Thahzi with log4j and I see that the loki timestamp always has 000000 in the micro/nanoseconds part.
log4j is able to give the timestamps with finer resolution. I can put these into the text part. But still it does not reach the loki timestamp.

**Comment by tkowalcz** [comment]
You are right that log4j supports finer resolution and Tjahzi should support it but currently doesn't. I'll fix it.

**Comment by tkowalcz** [comment, marked as answer]
I have released version [0.9.24](https://github.com/tkowalcz/tjahzi/releases/tag/tjahzi-parent-0.9.24) that fixes this.

**Reply by thgau** [reply]
👍 Thank you very much. That was a really fast reaction. And it is very helpful to see loki entries from different bins now in the correct order
